### ISSUE TITLE
Proposal for new a11y rule to disallow empty anchor elements

### ISSUE THREAD

**Issue by drewlee** [ember-template-lint/ember-template-lint#2113]
Currently, ember-template-lint disallows unhelpful link text such as "click here" via the [no-invalid-link-text](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-invalid-link-text.md) rule. However, there is nothing in place to prevent empty link text, which is even more harmful. When a link contains no content, the link's `href` ends up as the accessible name. This results in a poor experience and requires the user to resort to trial and error to determine the purpose of the link.

This proposes the authoring of a new rule that disallows empty anchor elements and `LinkTo` helpers. Hidden links should be ignored, while alternatively labeled links (using `aria-label` or `aria-labelledby`) should pass. This is essentially a port of JSXs A11y [anchor-has-content](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-has-content.md) rule.

**Comment by bmish** [comment]
Seems like the existing rule `no-invalid-link-text` could be updated to enforce this additional check?

**Comment by drewlee** [comment]
Yes, that may likely be the case. We'll need to dig around and determine whether `no-invalid-link-text` can be simply augmented.

**Comment by MelSumner** [comment]
@drewlee @bmish a little late to this party, sorry, but doesn't `https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/link-href-attributes.md` already cover this issue? I don't think it's a problem that it's now covered in two ways, though.

**Comment by bmish** [comment]
Hmm, good point. It looks like the logic in each check might be slightly different. Not too big of a deal but also not great to have redundancy.

**Comment by drewlee** [comment]
Those aren't really the same thing though. This disallows links with no text content `<a href="https://foo"></a>`, while `link-href-attributes` disallows links without an `href` attribute.

**Comment by drewlee** [comment]
Here's a better way to illustrate the differences between the two. The following is output given some simple markup related to the issues we're guarding against.

Give the template snippet `<a href="https://example.com"></a>`:

Passes for "link-href-attributes"
```bash
ember-template-lint --no-config-path app/templates/link-template.hbs --rule 'link-href-attributes'
```

Fails for "no-invalid-link-text"
```bash
ember-template-lint --no-config-path app/templates/link-template.hbs --rule 'no-invalid-link-text:["error", {"allowEmptyLinks": false}]'
app/templates/link-template.hbs
  1:0  error  Links should have descriptive text  no-invalid-link-text
```

Give the template snippet `<a>https://example.com</a>`:
Fails for "link-href-attributes"
```bash
ember-template-lint --no-config-path app/templates/link-template.hbs --rule 'link-href-attributes:error'
app/templates/link-template.hbs
  1:0  error  a tags must have an href attribute  link-href-attributes
```

Passes for "no-invalid-link-text"
```bash
ember-template-lint --no-config-path app/templates/link-template.hbs --rule 'no-invalid-link-text:["error", {"allowEmptyLinks": false}]'
```
